samples/outputs/sample_004.json

```diff
@@ -154,7 +154,53 @@
                     "max_nested_conditional": 0
                 }
             }
-        ]  
+        ],
+        "definition_lines": [
+            [
+                1,
+                6
+            ],
+            [
+                3,
+                2
+            ],
+            [
+                11,
+                0
+            ],
+            [
+                13,
+                2
+            ],
+            [
+                14,
+                0
+            ],
+            [
+                18,
+                0
+            ],
+            [
+                21,
+                0
+            ],
+            [
+                23,
+                0
```

Comment:
if you look at sample_004.js you'd find that line 23 is `function a () { function b () {} function *c () {} class D {} return }`. It has multiple functions and one class, but the definition_lines only has one [23,0] because it's a set. Should the output contain four of [23,0] to represent that it has multiple functions here or the goal for the definitions list here is different?

Reply:
That's an interesting question. We can see that this information is missing data for us to identify _what_ is being declared _where_. Even if we have the 4 `[23,0]` it's not very useful because we can't link it back to anything, really.

Added to that is the fact that this is not even used anywhere at this point: https://github.com/codecov/worker/blob/5e352be2189b197547c5370d7c70a707bc20c447/services/static_analysis/single_file_analyzer.py#L54 , and I don't really know what this would be used for, really.

However if we want a list with _all_ declaration lines, then we should include _all_ declaration lines. I would change the output to contain all the declarations (so 4 [23, 0]). That is my opinion, but there's no right or wrong in this case.

What do you think?